app/controllers/employees_controller.rb

```diff
@@ -0,0 +1,73 @@
+class EmployeesController < ApplicationController
+  load_and_authorize_resource
+
+  def new
+    @employee = Employee.new
+    @company = Company.find(params[:company_id])
+    @shop = Shop.find(params[:shop_id])
+  end
+
+  def create
+    @company = Company.find(params[:company_id])
+    @shop = Shop.find(params[:shop_id])
+    @user = User.find(params[:employee][:user_id])
+    @employee = Employee.new(employee_params.merge(user_id: @user.id,
+                             shop_id: @shop.id, company_id: @company.id,
+                             address_id: @user.address.id))
+    if @employee.save
+      redirect_to company_shop_employee_path(@company, @shop, @employee)
+      flash[:success] = 'Employee has been created!'
+    else
+      render 'new'
+    end
+  end
+
+  def show
+    @company = Company.find(params[:company_id])
+    @employee = Employee.find(params[:id])
+    @shop = Shop.find(params[:shop_id])
+  end
+
+  def destroy
+    @company = Company.find(params[:company_id])
+    @shop = Company.find(params[:shop_id])
+    @employee = Employee.find(params[:id])
+    flash[:success] = 'Employee has been deleted!'
+    redirect_to company_shop_path(@company, @shop)
+    @employee.destroy
+  end
+
+  def index
+    @company = Company.find(params[:company_id])
+    @shop = Shop.find(params[:shop_id])
+    @employees = Employee.all
```

Comment:
this can be loaded with load_and_authorize_resource (@Company and @shop and @employee)